anndata/_core/index.py

```diff
@@ -127,8 +127,14 @@ def _subset(a: Union[np.ndarray, pd.DataFrame], subset_idx: Index):
 @_subset.register(DaskArray)
 def _subset_dask(a: DaskArray, subset_idx: Index):
     if all(isinstance(x, cabc.Iterable) for x in subset_idx):
-        subset_idx = np.ix_(*subset_idx)
-        return a.vindex[subset_idx]
+        if isinstance(a._meta, csc_matrix):
+            return a[:, subset_idx[1]][subset_idx[0], :]
+        elif isinstance(a._meta, spmatrix):
+            return a[subset_idx[0], :][:, subset_idx[1]]
+        else:
+            # TODO: this may have been working for some cases?
+            subset_idx = np.ix_(*subset_idx)
+            return a.vindex[subset_idx]
```

Comment:
Ideally the `vindex` case would be more efficient by doing indexing across both dimensions at the same time. Though I haven't actually checked this.

But I believe there were some test cases passing without this change which are now going through the new branches. It could be worth making those cases go through `vindex` instead of `[obs_idx, :][:, var_idx]`